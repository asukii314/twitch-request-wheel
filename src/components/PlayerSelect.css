button.change-col {
    text-transform: uppercase;
    margin: 3px;
    font-size: 11px;
    background-color: lightgray;
}

div.change-col-buttons-container {
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    visibility: hidden;
    margin-top: -25px;
}

div.change-col-buttons-container:hover {
    display: flex;
    flex-direction: row;
}

div.change-col-buttons-container .change-col {
    background-color: tomato;
    border-radius: 1rem;
}
div.change-col-buttons-container .change-col.send-code {
    background-color: #d3d3d3;
}
div.change-col-buttons-container .change-col.send-code:hover {
    background-color: #e3e3e3;
}

.player-card:hover {
    background-color: dimgray;
}

.player-card:hover > .change-col-buttons-container {
    visibility: visible;
}

.player-count {
    font-size: 16px;
    margin-top: 5px;
}

.player-count.overlimit {
    font-weight: bold;
    color: darkorange;
}

#main-screen button.btn.start-game {
    padding-top: 0.5rem;
    padding-bottom: 0.5rem;
}
#main-screen .card-header-item > .input-group + button.start-game {
    padding-top: 1px;
    padding-bottom: 1px;
}

button.start-game,
.card-header-item > .input-group {
    width: 100%;
    margin-bottom: 5px;
}

input.room-code::placeholder {
    /* letter-spacing: -0.5px; */
    font-size: 90%;
}

.player-select .player-card-container {
    overflow-y: auto;
    height: calc(100vh - 250px);
    font-size: 16px;
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.player-card-column {
    width: 50%;
    height: 100%;
    margin: 0 5px;
    border-radius: 5px;
}

.player-card-column.interested {
    background-color: slategrey;
}

.player-card-column.playing {
    background-color: teal;
}

.player-card-column.joined {
    background-color: seagreen;
    display: none;
}

.player-card-column > p {
    margin-top: 0;
    padding-top: 10px;
    margin-bottom: 10px;
    font-size: 17px;
    font-weight: bold;
}

.player-card {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    background-color: gray;
    margin: 8px;
    border-radius: 1rem;
    border-style: outset;
    border-color: cadetblue;
}

.player-card-username {
    display: flex;
}

.player-card-username > img {
    width: 12px;
    margin-left: 2px;
}

.player-name {
    margin: 0px 0px 0px 16px;
    font-weight: 500;
    word-wrap: break-word;
    text-align: left;
}

img + p.player-name {
    margin-left: 0px;
}

button.dice {
    width: 25px;
    height: 25px;
    background-color: greenyellow;
    border-radius: 5px;
    margin-left: 10px;
    padding: 0px;
    vertical-align: middle;
    cursor: pointer;
}

button.dice > img {
    width: 100%;
    height: 100%;
    vertical-align: unset;
}

.toggle-streamer-seat,
.card-header-item {
    display: flex;
    align-items: center;
}

.toggle-streamer-seat .toggle-label,
.card-header-item-label {
    font-size: 12px;
    max-width: 40px;
    overflow-wrap: break-word;
    text-align: center;
}
.toggle-streamer-seat .toggle-label {
    max-width: 80px;
    text-align: right;
}

.toggle-streamer-seat .form-switch {
    margin-left: 8px;
    padding-left: 3em;
}
.toggle-streamer-seat .form-switch .form-check-input {
    height: 1.6em;
    width: 3em;
    margin-left: -3em;
    background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='-4 -4 8 8'%3e%3ccircle r='3' fill='%23fff'/%3e%3c/svg%3e");
    background-position: left center;
    border-radius: 3em;
    transition: background-position 0.15s ease-in-out;
    background-color: #f87171;
    border-color: #f87171;
}
.toggle-streamer-seat .form-switch .form-check-input:focus {
    background-image: url("data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' viewBox='-4 -4 8 8'%3e%3ccircle r='3' stroke='%2386b7fe' stroke-width='0.5' fill='%23fdd'/%3e%3c/svg%3e");
}
.toggle-streamer-seat .form-check-input:checked {
    background-color: #34d399;
    border-color: #34d399;
}
input#reserve-seat-for-streamer:after {
    display: block;
    content: "no";
    position: absolute;
    color: #fff;
    font-size: 0.7rem;
    text-align: center;
    width: 2.25em;
    line-height: 2em;
    font-weight: bold;
    margin-left: 1.75em;
}
input#reserve-seat-for-streamer:checked:after {
    content: "yes";
    margin-left: 0;
}
