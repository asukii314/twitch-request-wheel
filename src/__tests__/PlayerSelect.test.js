import {createRenderer} from 'react-test-renderer/shallow';
import PlayerSelect from '../components/PlayerSelect';
import React from 'react';

describe('PlayerSelect', () => {

    const propsBlather = {
        game: {
            name: 'Blather \'Round',
            longName: 'Blather \'Round (Party Pack 7)',
            partyPack: 'Party Pack 7',
            'Min players': 2,
            'Max players': 6,
            Variants: [
                'blather',
                'blather round',
                'blather \'round',
                'blatherround',
                'blatheround'
            ],
            override: true
        },
        userLookup: {}
    };
    const propsTMP2 = {
        game: {
            name: 'Trivia Murder Party 2',
            longName: 'Trivia Murder Party 2 (Party Pack 6)',
            partyPack: 'Party Pack 6',
            'Min players': 1,
            'Max players': 8,
            Variants: [
                'tmp2',
                'tmp 2',
                'trivia murder party 2'
            ],
            username: 'dcpesses',
            time: 1627401634507,
            locked: false,
            chosen: false,
            override: false
        },
        userLookup: {}
    };
    const state = {
        interested: [
            {username: 'player4'},
            {username: 'player5', isPrioritySeat: true}
        ],
        playing: [
            {username: 'player1', isPrioritySeat: true},
            {username: 'player2', isPrioritySeat: true},
            {username: 'player3', isPrioritySeat: false},
            {username: 'player6', isPrioritySeat: false},
            {username: 'player7', isPrioritySeat: false},
            {username: 'player8'},
            {username: 'player9'}
        ],
        joined: [],
        streamerSeat: false,
        isQueueOpen: true,
        columnWidth: 524
    };

    describe('componentDidMount', () => {
        test('should create event listener and call updateColumnSizes', () => {
            jest.spyOn(window, 'addEventListener').mockImplementation(() => {});
            const component = new PlayerSelect(propsTMP2);
            jest.spyOn(component, 'updateColumnSizes').mockImplementation(()=>{});
            component.componentDidMount();
            expect(component.updateColumnSizes).toHaveBeenCalledTimes(1);
            expect(window.addEventListener).toHaveBeenCalledWith('resize', component.updateColumnSizes);
        });
    });

    describe('componentWillUnmount', () => {
        test('should remove event listener', () => {
            const component = new PlayerSelect(propsTMP2);
            jest.spyOn(window, 'removeEventListener').mockImplementation(() => {});
            component.componentWillUnmount();
            expect(window.removeEventListener).toHaveBeenCalledWith('resize', component.updateColumnSizes);
        });
    });

    describe('updateColumnSizes', () => {
        test('should update the column width', () => {
            const component = new PlayerSelect(propsTMP2);
            component.state = state;
            component.firstColumn = {
                current: {
                    offsetWidth: 100
                }
            };
            jest.spyOn(component, 'setState').mockImplementation(()=>{});

            component.updateColumnSizes();

            expect(component.setState).toHaveBeenCalledTimes(1);
            expect(component.setState.mock.calls[0][0](component.state)).toEqual({
                ...component.state,
                columnWidth: 100
            });
        });
    });

    describe('handleNewPlayerRequest', () => {
        const component = new PlayerSelect(propsTMP2);

        jest.spyOn(component, 'setState').mockImplementation(()=>{});
        jest.spyOn(component, 'removeUser').mockImplementation(()=>{});

        beforeEach(() => {
            component.state = {
                ...state,
                interested: [
                    {username: 'player4'},
                    {username: 'player5', isPrioritySeat: true}
                ],
                playing: [
                    {username: 'player1', isPrioritySeat: true},
                    {username: 'player2', isPrioritySeat: false},
                    {username: 'player3'}
                ],
                joined: [
                    {username: 'player9'}
                ],
            };
            jest.resetAllMocks();
        });

        test('should return the expected message when successful', () => {
            component.state.isQueueOpen = true;
            expect(component.handleNewPlayerRequest('player9', {})).toBe(
                'you are already in the lobby.'
            );
            expect(component.handleNewPlayerRequest('player10', {})).toBe(
                'you have successfully joined the lobby.'
            );
            expect(component.handleNewPlayerRequest('player11', {isPrioritySeat: true})).toBe(
                'you have been successfully added to the lobby.'
            );
            component.state.isQueueOpen = false;
            expect(component.handleNewPlayerRequest('player10', {})).toBe(
                'the queue is currently closed; users have already been selected for this game.'
            );
            expect(component.handleNewPlayerRequest('player11', {isPrioritySeat: true})).toBe(
                'you have been successfully added to the lobby.'
            );

            expect(component.removeUser).toHaveBeenCalledTimes(3);
            expect(component.setState).toHaveBeenCalledTimes(3);
            expect(component.setState.mock.calls[1][0](component.state)).toEqual({
                ...component.state,
                playing: [
                    ...component.state.playing,
                    {username: 'player11', isPrioritySeat: true}
                ]
            });
        });
        test('should return the expected message when an error occurs', () => {
            component.state.playing = null;
            component.state.interested = null;
            component.state.isQueueOpen = true;

            expect(component.handleNewPlayerRequest('player9', {})).toBe(
                'you are already in the lobby.'
            );
            expect(component.handleNewPlayerRequest('player10', {isPrioritySeat: true})).toBe(
                'there was an error adding you to the lobby.'
            );
            // component.state.isQueueOpen = false;
            expect(component.handleNewPlayerRequest('player11', {})).toBe(
                'there was an error adding you to the lobby.'
            );


            expect(component.removeUser).toHaveBeenCalledTimes(0);
            expect(component.setState).toHaveBeenCalledTimes(0);

        });
    });

    describe('removeUser', () => {
        test('should call setState', () => {
            const component = new PlayerSelect(propsTMP2);
            jest.spyOn(component, 'setState').mockImplementation(()=>{});
            component.state = {
                ...state,
                interested: [
                    {username: 'CrunchyButtMD'},
                    {username: 'HyalineRose'}
                ],
                playing: [
                    {username: 'Iniquity_Stepbro'},
                    {username: 'HiddenPudding'}
                ],
                joined: [
                    {username: 'Aurora88877'}
                ]
            };
            component.removeUser('HyalineRose');
            expect(component.setState).toHaveBeenCalledTimes(1);
            expect(component.setState.mock.calls[0][0](component.state)).toEqual({
                ...component.state,
                interested: [
                    {username: 'CrunchyButtMD'}
                ]
            });
        });
    });

    describe('clearQueue', () => {
        test('should clear the users from state', () => {
            const component = new PlayerSelect(propsTMP2);
            component.state = state;
            jest.spyOn(component, 'setState').mockImplementation(()=>{});

            component.clearQueue();

            expect(component.setState).toHaveBeenCalledTimes(1);
            expect(component.setState.mock.calls[0][0](component.state)).toEqual({
                ...component.state,
                interested: [],
                playing: [],
                joined: []
            });
        });
    });

    describe('openQueue', () => {
        test('should set an open queue state', () => {
            const component = new PlayerSelect(propsTMP2);
            component.state = state;
            jest.spyOn(component, 'setState').mockImplementation(()=>{});

            component.openQueue();

            expect(component.setState).toHaveBeenCalledTimes(1);
            expect(component.setState.mock.calls[0][0](component.state)).toEqual({
                ...component.state,
                isQueueOpen: true
            });
        });
    });

    describe('closeQueue', () => {
        test('should set a closed queue state', () => {
            const component = new PlayerSelect(propsTMP2);
            component.state = state;
            jest.spyOn(component, 'setState').mockImplementation(()=>{});

            component.closeQueue();

            expect(component.setState).toHaveBeenCalledTimes(1);
            expect(component.setState.mock.calls[0][0](component.state)).toEqual({
                ...component.state,
                isQueueOpen: false
            });
        });
    });

    describe('playerCount', () => {
        const component = new PlayerSelect(propsTMP2);
        component.state = state;

        test('should return number of players that will be playing, including streamer', () => {
            component.state.streamerSeat = true;
            expect(component.playerCount()).toBe(8);
        });
        test('should return number of players that will be playing, excluding streamer', () => {
            component.state.streamerSeat = false;
            expect(component.playerCount()).toBe(7);
        });
    });


    describe('toggleStreamerSeat', () => {
        test('should toggle the streamer seat state', () => {
            const component = new PlayerSelect(propsTMP2);
            component.state = state;
            component.state.streamerSeat = false;
            jest.spyOn(component, 'setState').mockImplementation(()=>{});

            component.toggleStreamerSeat();

            expect(component.setState).toHaveBeenCalledTimes(1);
            expect(component.setState.mock.calls[0][0](component.state)).toEqual({
                ...component.state,
                streamerSeat: true
            });
        });
    });

    describe('startGame', () => {
        test('should call props.startGame', () => {
            const component = new PlayerSelect(
                Object.assign({}, propsTMP2, {startGame: jest.fn()})
            );
            jest.spyOn(component, 'setState').mockImplementation(()=>{});

            let prevState = component.state;
            component.startGame();

            expect(component.setState).toHaveBeenCalledTimes(1);
            expect(component.setState.mock.calls[0][0](component.state)).toEqual({
                ...component.state,
                interested: [],
                playing: [],
                joined: [],
                roomCode: null,
                undoState: prevState
            });
            expect(component.props.startGame).toHaveBeenCalledTimes(1);
        });
    });

    describe('randomizePlayers', () => {
        test('should test all cases and branches', () => {
            // const mockMath = Object.create(global.Math);
            jest.spyOn(global.Math, 'floor')
                .mockImplementationOnce(()=>4)
                .mockImplementationOnce(()=>0)
                .mockImplementationOnce(()=>4)
                .mockImplementation(()=>2);

            const component = new PlayerSelect(propsBlather);
            component.state = {
                ...state,
                interested: [
                    {username: 'player4'},
                    {username: 'player5'},
                    {username: 'player6'},
                    {username: 'player7'},
                    {username: 'player8'}
                ],
                playing: [
                    {username: 'player1', isPrioritySeat: true},
                    {username: 'player2', isPrioritySeat: true},
                    {username: 'player3'}
                ],
            };
            jest.spyOn(component, 'setState').mockImplementation(()=>{});

            component.randomizePlayers();

            expect(component.setState).toHaveBeenCalledTimes(1);

            expect(component.setState.mock.calls[0][0](component.state).interested.length).toBe(2);
            expect(component.setState.mock.calls[0][0](component.state).playing.length).toBe(6);
        });
    });

    describe('render', () => {
        test('should render component', () => {
            const shallowRenderer = createRenderer();
            shallowRenderer.render(<PlayerSelect />);
            let component = shallowRenderer.getRenderOutput();
            expect(component).toMatchSnapshot();
            expect(component.type).toBe('div');
            shallowRenderer.unmount();
        });
        test('should render component using Blather data', () => {
            const shallowRenderer = createRenderer();
            shallowRenderer.render(<PlayerSelect game={propsBlather.game} userLookup={propsBlather.userLookup} />);
            let instance = shallowRenderer.getMountedInstance();
            instance.setState(state);
            let component = shallowRenderer.getRenderOutput();
            expect(component).toMatchSnapshot();
            shallowRenderer.unmount();
        });
        test('should render component using TMP2 data', () => {
            const shallowRenderer = createRenderer();
            shallowRenderer.render(<PlayerSelect game={propsTMP2.game} userLookup={propsTMP2.userLookup} />);
            let instance = shallowRenderer.getMountedInstance();
            instance.setState(state);
            expect(shallowRenderer.getRenderOutput()).toMatchSnapshot();
            shallowRenderer.unmount();
        });
    });
});
