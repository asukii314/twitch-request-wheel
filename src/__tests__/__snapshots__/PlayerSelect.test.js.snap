// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerSelect render should render component 1`] = `
<div
  className="card player-select-container"
>
  <div
    className="card-header d-flex justify-content-between"
  >
    <div
      className="toggle-streamer-seat card-header-item"
    >
      <label
        className="toggle-label form-check-label"
        htmlFor="reserve-seat-for-streamer"
      >
        Reserve seat for streamer?
      </label>
      <div
        className="form-check form-switch"
      >
        <input
          className="form-check-input"
          defaultChecked={false}
          id="reserve-seat-for-streamer"
          onChange={[Function]}
          role="switch"
          type="checkbox"
        />
      </div>
    </div>
    <div
      className="fs-2 lh-sm game-name"
    >
      <b>
        TBD
      </b>
      <div
        className="player-count"
      >
        0
         of 
         seats claimed
      </div>
    </div>
    <div
      className="card-header-item d-flex align-items-center align-self-center flex-column"
    >
      <button
        className="btn btn-sm start-game"
        disabled={true}
        onClick={[Function]}
      >
        Start Game
      </button>
    </div>
  </div>
  <div
    className="card-body player-card-container"
  >
    <div
      className="player-card-column interested"
    >
      <p
        className="player-card-column-header"
      >
        Interested
      </p>
    </div>
    <div
      className="player-card-column playing"
    >
      <p
        className="player-card-column-header"
      >
        Playing
        <button
          className="dice"
          onClick={[Function]}
        >
          <img
            alt="dice icon"
            src="file-stub"
          />
        </button>
      </p>
    </div>
  </div>
</div>
`;

exports[`PlayerSelect render should render component using Blather data 1`] = `
<div
  className="card player-select-container"
>
  <div
    className="card-header d-flex justify-content-between"
  >
    <div
      className="toggle-streamer-seat card-header-item"
    >
      <label
        className="toggle-label form-check-label"
        htmlFor="reserve-seat-for-streamer"
      >
        Reserve seat for streamer?
      </label>
      <div
        className="form-check form-switch"
      >
        <input
          className="form-check-input"
          defaultChecked={false}
          id="reserve-seat-for-streamer"
          onChange={[Function]}
          role="switch"
          type="checkbox"
        />
      </div>
    </div>
    <div
      className="fs-2 lh-sm game-name"
    >
      <b>
        Blather 'Round
      </b>
      <div
        className="player-count overlimit"
      >
        7
         of 
        6
         seats claimed
      </div>
    </div>
    <div
      className="card-header-item d-flex align-items-center align-self-center flex-column"
    >
      <button
        className="btn btn-sm start-game"
        disabled={true}
        onClick={[Function]}
      >
        Start Game
      </button>
    </div>
  </div>
  <div
    className="card-body player-card-container"
  >
    <div
      className="player-card-column interested"
    >
      <p
        className="player-card-column-header"
      >
        Interested
      </p>
      <div
        className="player-card lh-sm fs-5"
      >
        <div
          className="player-card-username"
        >
          <img
            alt="Priority seat redemption"
            src="file-stub"
          />
          <p
            className="player-name"
            style={
              Object {
                "maxWidth": 499,
              }
            }
          >
            player5
          </p>
        </div>
        <div
          className="change-col-buttons-container"
        >
          <button
            className="change-col"
            onClick={[Function]}
          >
            Playing
          </button>
          <button
            className="change-col"
            onClick={[Function]}
          >
            X
          </button>
        </div>
      </div>
      <div
        className="player-card lh-sm fs-5"
      >
        <div
          className="player-card-username"
        >
          <p
            className="player-name"
            style={
              Object {
                "maxWidth": 499,
              }
            }
          >
            player4
          </p>
        </div>
        <div
          className="change-col-buttons-container"
        >
          <button
            className="change-col"
            onClick={[Function]}
          >
            Playing
          </button>
          <button
            className="change-col"
            onClick={[Function]}
          >
            X
          </button>
        </div>
      </div>
    </div>
    <div
      className="player-card-column playing"
    >
      <p
        className="player-card-column-header"
      >
        Playing
        <button
          className="dice"
          onClick={[Function]}
        >
          <img
            alt="dice icon"
            src="file-stub"
          />
        </button>
      </p>
      <div
        className="player-card lh-sm fs-5"
      >
        <div
          className="player-card-username"
        >
          <img
            alt="Priority seat redemption"
            src="file-stub"
          />
          <p
            className="player-name"
            style={
              Object {
                "maxWidth": 499,
              }
            }
          >
            player1
          </p>
        </div>
        <div
          className="change-col-buttons-container"
        >
          <button
            className="change-col"
            onClick={[Function]}
          >
            Interested
          </button>
          <button
            className="change-col"
            onClick={[Function]}
          >
            X
          </button>
        </div>
      </div>
      <div
        className="player-card lh-sm fs-5"
      >
        <div
          className="player-card-username"
        >
          <img
            alt="Priority seat redemption"
            src="file-stub"
          />
          <p
            className="player-name"
            style={
              Object {
                "maxWidth": 499,
              }
            }
          >
            player2
          </p>
        </div>
        <div
          className="change-col-buttons-container"
        >
          <button
            className="change-col"
            onClick={[Function]}
          >
            Interested
          </button>
          <button
            className="change-col"
            onClick={[Function]}
          >
            X
          </button>
        </div>
      </div>
      <div
        className="player-card lh-sm fs-5"
      >
        <div
          className="player-card-username"
        >
          <p
            className="player-name"
            style={
              Object {
                "maxWidth": 499,
              }
            }
          >
            player3
          </p>
        </div>
        <div
          className="change-col-buttons-container"
        >
          <button
            className="change-col"
            onClick={[Function]}
          >
            Interested
          </button>
          <button
            className="change-col"
            onClick={[Function]}
          >
            X
          </button>
        </div>
      </div>
      <div
        className="player-card lh-sm fs-5"
      >
        <div
          className="player-card-username"
        >
          <p
            className="player-name"
            style={
              Object {
                "maxWidth": 499,
              }
            }
          >
            player6
          </p>
        </div>
        <div
          className="change-col-buttons-container"
        >
          <button
            className="change-col"
            onClick={[Function]}
          >
            Interested
          </button>
          <button
            className="change-col"
            onClick={[Function]}
          >
            X
          </button>
        </div>
      </div>
      <div
        className="player-card lh-sm fs-5"
      >
        <div
          className="player-card-username"
        >
          <p
            className="player-name"
            style={
              Object {
                "maxWidth": 499,
              }
            }
          >
            player7
          </p>
        </div>
        <div
          className="change-col-buttons-container"
        >
          <button
            className="change-col"
            onClick={[Function]}
          >
            Interested
          </button>
          <button
            className="change-col"
            onClick={[Function]}
          >
            X
          </button>
        </div>
      </div>
      <div
        className="player-card lh-sm fs-5"
      >
        <div
          className="player-card-username"
        >
          <p
            className="player-name"
            style={
              Object {
                "maxWidth": 499,
              }
            }
          >
            player8
          </p>
        </div>
        <div
          className="change-col-buttons-container"
        >
          <button
            className="change-col"
            onClick={[Function]}
          >
            Interested
          </button>
          <button
            className="change-col"
            onClick={[Function]}
          >
            X
          </button>
        </div>
      </div>
      <div
        className="player-card lh-sm fs-5"
      >
        <div
          className="player-card-username"
        >
          <p
            className="player-name"
            style={
              Object {
                "maxWidth": 499,
              }
            }
          >
            player9
          </p>
        </div>
        <div
          className="change-col-buttons-container"
        >
          <button
            className="change-col"
            onClick={[Function]}
          >
            Interested
          </button>
          <button
            className="change-col"
            onClick={[Function]}
          >
            X
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PlayerSelect render should render component using TMP2 data 1`] = `
<div
  className="card player-select-container"
>
  <div
    className="card-header d-flex justify-content-between"
  >
    <div
      className="toggle-streamer-seat card-header-item"
    >
      <label
        className="toggle-label form-check-label"
        htmlFor="reserve-seat-for-streamer"
      >
        Reserve seat for streamer?
      </label>
      <div
        className="form-check form-switch"
      >
        <input
          className="form-check-input"
          defaultChecked={false}
          id="reserve-seat-for-streamer"
          onChange={[Function]}
          role="switch"
          type="checkbox"
        />
      </div>
    </div>
    <div
      className="fs-2 lh-sm game-name"
    >
      <b>
        Trivia Murder Party 2
      </b>
      <div
        className="player-count"
      >
        7
         of 
        8
         seats claimed
      </div>
    </div>
    <div
      className="card-header-item d-flex align-items-center align-self-center flex-column"
    >
      <button
        className="btn btn-sm start-game"
        disabled={false}
        onClick={[Function]}
      >
        Start Game
      </button>
    </div>
  </div>
  <div
    className="card-body player-card-container"
  >
    <div
      className="player-card-column interested"
    >
      <p
        className="player-card-column-header"
      >
        Interested
      </p>
      <div
        className="player-card lh-sm fs-5"
      >
        <div
          className="player-card-username"
        >
          <img
            alt="Priority seat redemption"
            src="file-stub"
          />
          <p
            className="player-name"
            style={
              Object {
                "maxWidth": 499,
              }
            }
          >
            player5
          </p>
        </div>
        <div
          className="change-col-buttons-container"
        >
          <button
            className="change-col"
            onClick={[Function]}
          >
            Playing
          </button>
          <button
            className="change-col"
            onClick={[Function]}
          >
            X
          </button>
        </div>
      </div>
      <div
        className="player-card lh-sm fs-5"
      >
        <div
          className="player-card-username"
        >
          <p
            className="player-name"
            style={
              Object {
                "maxWidth": 499,
              }
            }
          >
            player4
          </p>
        </div>
        <div
          className="change-col-buttons-container"
        >
          <button
            className="change-col"
            onClick={[Function]}
          >
            Playing
          </button>
          <button
            className="change-col"
            onClick={[Function]}
          >
            X
          </button>
        </div>
      </div>
    </div>
    <div
      className="player-card-column playing"
    >
      <p
        className="player-card-column-header"
      >
        Playing
        <button
          className="dice"
          onClick={[Function]}
        >
          <img
            alt="dice icon"
            src="file-stub"
          />
        </button>
      </p>
      <div
        className="player-card lh-sm fs-5"
      >
        <div
          className="player-card-username"
        >
          <img
            alt="Priority seat redemption"
            src="file-stub"
          />
          <p
            className="player-name"
            style={
              Object {
                "maxWidth": 499,
              }
            }
          >
            player1
          </p>
        </div>
        <div
          className="change-col-buttons-container"
        >
          <button
            className="change-col"
            onClick={[Function]}
          >
            Interested
          </button>
          <button
            className="change-col"
            onClick={[Function]}
          >
            X
          </button>
        </div>
      </div>
      <div
        className="player-card lh-sm fs-5"
      >
        <div
          className="player-card-username"
        >
          <img
            alt="Priority seat redemption"
            src="file-stub"
          />
          <p
            className="player-name"
            style={
              Object {
                "maxWidth": 499,
              }
            }
          >
            player2
          </p>
        </div>
        <div
          className="change-col-buttons-container"
        >
          <button
            className="change-col"
            onClick={[Function]}
          >
            Interested
          </button>
          <button
            className="change-col"
            onClick={[Function]}
          >
            X
          </button>
        </div>
      </div>
      <div
        className="player-card lh-sm fs-5"
      >
        <div
          className="player-card-username"
        >
          <p
            className="player-name"
            style={
              Object {
                "maxWidth": 499,
              }
            }
          >
            player3
          </p>
        </div>
        <div
          className="change-col-buttons-container"
        >
          <button
            className="change-col"
            onClick={[Function]}
          >
            Interested
          </button>
          <button
            className="change-col"
            onClick={[Function]}
          >
            X
          </button>
        </div>
      </div>
      <div
        className="player-card lh-sm fs-5"
      >
        <div
          className="player-card-username"
        >
          <p
            className="player-name"
            style={
              Object {
                "maxWidth": 499,
              }
            }
          >
            player6
          </p>
        </div>
        <div
          className="change-col-buttons-container"
        >
          <button
            className="change-col"
            onClick={[Function]}
          >
            Interested
          </button>
          <button
            className="change-col"
            onClick={[Function]}
          >
            X
          </button>
        </div>
      </div>
      <div
        className="player-card lh-sm fs-5"
      >
        <div
          className="player-card-username"
        >
          <p
            className="player-name"
            style={
              Object {
                "maxWidth": 499,
              }
            }
          >
            player7
          </p>
        </div>
        <div
          className="change-col-buttons-container"
        >
          <button
            className="change-col"
            onClick={[Function]}
          >
            Interested
          </button>
          <button
            className="change-col"
            onClick={[Function]}
          >
            X
          </button>
        </div>
      </div>
      <div
        className="player-card lh-sm fs-5"
      >
        <div
          className="player-card-username"
        >
          <p
            className="player-name"
            style={
              Object {
                "maxWidth": 499,
              }
            }
          >
            player8
          </p>
        </div>
        <div
          className="change-col-buttons-container"
        >
          <button
            className="change-col"
            onClick={[Function]}
          >
            Interested
          </button>
          <button
            className="change-col"
            onClick={[Function]}
          >
            X
          </button>
        </div>
      </div>
      <div
        className="player-card lh-sm fs-5"
      >
        <div
          className="player-card-username"
        >
          <p
            className="player-name"
            style={
              Object {
                "maxWidth": 499,
              }
            }
          >
            player9
          </p>
        </div>
        <div
          className="change-col-buttons-container"
        >
          <button
            className="change-col"
            onClick={[Function]}
          >
            Interested
          </button>
          <button
            className="change-col"
            onClick={[Function]}
          >
            X
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
